src/python_testing/TC_GCAST_2_3.py

```diff
@@ -40,12 +40,92 @@
 from mobly import asserts
 from matter.testing.event_attribute_reporting import AttributeSubscriptionHandler
 import matter.clusters as Clusters
-from matter.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from matter.testing.matter_testing import AttributeMatcher, MatterBaseTest, TestStep, async_test_body, default_matter_test_main, has_cluster, run_if_endpoint_matches
 from matter.interaction_model import InteractionModelError, Status
 
+from src.python_testing.TC_GCAST_2_1 import is_groupcast_supporting_cluster
```

Comment:
This is odd. I did not expect this type of import. Suggest making a TC_GCAST_common.py file for these utilities.